pkg/reconciler/reconciler.go

```diff
@@ -146,6 +146,8 @@ type RootOptions struct {
 
 // Run configures and starts the various components of a reconciler process.
 func Run(opts Options) {
+	// Start listening to signals
+	signalCtx := signals.SetupSignalHandler()
```

Comment:
Mostly lgtm, just curious why the SetupSignalHandler is moving to the front

Reply:
So that the context can be passed to `AnnotateStatusMode`